httpcore/_synchronization.py

```diff
@@ -83,10 +118,7 @@ async def __aexit__(
         exc_value: Optional[BaseException] = None,
         traceback: Optional[TracebackType] = None,
     ) -> None:
-        if self._backend == "trio":
-            self._trio_lock.release()
-        elif self._backend == "asyncio":
-            self._anyio_lock.release()
+        cast(_LockProto, self._lock).release()
```

Comment:
use `#type: ignore` to prevent runtime calls.

Reply:
Done